aws/internal/service/storagegateway/waiter/waiter.go

```diff
@@ -28,3 +30,41 @@ func StoredIscsiVolumeAvailable(conn *storagegateway.StorageGateway, volumeARN s
 
 	return nil, err
 }
+
+// NfsFileShareAvailable waits for a NFS File Share to return Available
+func NfsFileShareAvailable(conn *storagegateway.StorageGateway, fileShareArn string, timeout time.Duration) (*storagegateway.NFSFileShareInfo, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"BOOTSTRAPPING", "CREATING", "RESTORING", "UPDATING"},
+		Target:  []string{"AVAILABLE"},
+		Refresh: NfsFileShareStatus(conn, fileShareArn),
+		Timeout: timeout,
+		Delay:   NfsFileShareAvailableDelay,
+	}
+
+	outputRaw, err := stateConf.WaitForState()
+
+	if output, ok := outputRaw.(*storagegateway.NFSFileShareInfo); ok {
+		return output, err
+	}
+
+	return nil, err
+}
+
+func NfsFileShareNotFound(conn *storagegateway.StorageGateway, fileShareArn string, timeout time.Duration) (*storagegateway.NFSFileShareInfo, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending:        []string{"AVAILABLE", "DELETING", "FORCE_DELETING"},
+		Target:         []string{"NotFound"},
```

Comment:
Should work with just an empty array `[]string{}`.

Reply:
Done.